packages/vm/src/dynamic_link.rs

```diff
@@ -234,6 +236,56 @@ where
     Ok(write_to_contract(env, value)?.into())
 }
 
+pub fn native_validate_dynamic_link_interface<A: BackendApi, S: Storage, Q: Querier>(
+    env: &Environment<A, S, Q>,
+    address: u32,
+    interface: u32,
+) -> Result<u32, RuntimeError>
+where
+    A: BackendApi + 'static,
+    S: Storage + 'static,
+    Q: Querier + 'static,
+{
+    let contract_addr_raw = read_region(&env.memory(), address, 64)?;
+    let contract_addr: Addr = from_slice(&contract_addr_raw)
+        .map_err(|_| RuntimeError::new("Invalid contract address to validate interface"))?;
+    let expected_interface_binary = read_region(&env.memory(), interface, MAX_REGIONS_LENGTH)?;
+    let expected_interface: Vec<ExportType<FunctionType>> = from_slice(&expected_interface_binary)
+        .map_err(|_| RuntimeError::new("Invalid expected interface"))?;
+    let (module_result, gas_info) = env.api.get_wasmer_module(contract_addr.as_str());
+    process_gas_info::<A, S, Q>(env, gas_info)?;
+    let module = module_result.map_err(|_| RuntimeError::new("Cannot get module"))?;
+    let mut exported_fns: HashMap<String, FunctionType> = HashMap::new();
+    for f in module.exports().functions() {
+        exported_fns.insert(f.name().to_string(), f.ty().clone());
+    }
+
+    // No gas fee for comparison now
+    let mut err_msg = "The following functions are not implemented: ".to_string();
+    let mut is_err = false;
+    for expected_fn in expected_interface.iter() {
+        // if not expected
+        if !exported_fns
+            .get(expected_fn.name())
+            .map_or(false, |t| t == expected_fn.ty())
+        {
+            if is_err {
+                err_msg.push_str(", ");
+            };
+            err_msg.push_str(&format!("{}: {}", expected_fn.name(), expected_fn.ty()));
+            is_err = true;
+        }
+    }
+
+    if is_err {
+        // not expected
+        Ok(write_to_contract::<A, S, Q>(env, err_msg.as_bytes())?)
+    } else {
```

Comment:
nit: Is there a no problem without else state?

Reply:
L272?

Reply:
L272?

Reply:
L272? if it is, there are no problems. It is for adding ',' into the error message's tail before adding a new sentence _if it is not the first sentence_.

Reply:
I thought else statement is unnecessary, but I understood.

Reply:
The then clause needs to return the same typed value as the else clause. So this part needs to be one of as this PR or

```
if is_err {
    return Ok(write_to_contract...)
};
Ok(0)
```

And, I chose the style as this PR.